.gitignore

```diff
@@ -5,3 +5,5 @@ webgui/deploy.log
 webgui/env
 webgui/server_pid
 index.html
+Pipfile
+Pipfile.lock
```

Comment:
Removed these files because they always change and pollute diffs. It doesn't appear that they are used in deployment

Reply:
This may be moot if we dockerize, but pipenv can install from either requirements.txt or Pipfile. Requirements.txt needs to be periodically manually updated with a pipenv lock -r > requirements.txt, whereas the Pipfile reflects which modules have been currently added by pipenv. As long as the requirements.txt include locked version numbers, we can probably remove Pipfile and Pipfile.lock from version control.

Reply:
Ok I see now, I thought those files were just a byproduct.`requirements.txt` is easier to edit for humans so might be worth sticking with that. Usually I would have seen this sort file handled manually. I think for us the requirements file could be as short as:
```
-i https://pypi.org/simple

django==3.1.7
django-crispy-forms==1.11.2
pyzx==0.6.4
python-igraph==0.9.0
qiskit==0.24.0; python_version > 3.6

```
And let dependency resolution sort out the rest on each system, locking other packages as need arises.

But I'm not really sure about what the practices are with python projects.

Reply:
If you are unable to get pipenv on your server machine, the Pipfiles won't be very helpful I guess. It's just a statement of which sub-dependencies are used, for example, in qiskit, you wouldn't believe how many dependencies that one has, and the pipfiles show which ones were installed. When we choose another library to add to the project, pipenv will see which sub-dependencies are being used and if there is a conflict, it will try other possible combinations. If you install the packages one by one, you might end up with incompatible versions. It's not the end of the world if we can't use pipenv, ill add Pipfile and Pipfile.lock to gitignore if you haven't already. Locked versions and consistent ordering of libraries should be okay for now.

Reply:
Ok I propose a hybrid approach: We remove the pip files from the gitignore, but handle manually the requirements.txt, this way we have both human readable depndency and, as needed - even if not used in deployment,a complete snapshot in the Pipfiles.

Reply:
We just leave the rule that Pipfile changes allways deserve their own commit

Reply:
To keep the diffs nice